File: mycal.hs
multiplyBy :: Double -> Double -> Double
multiplyBy fac n = n * fac

square :: Double -> Double
square n = n * n

myCal :: Double -> Double -> Double
myCal = multiplyBy . square

fac :: Double
fac = 0.1

nums :: [Double]
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9]

transformedNums :: [Double]
transformedNums = map (\n -> myCal n fac) nums

main :: IO ()
main = do
  print transformedNums

File: mycal.js
const multiplyBy = (fac) => {
  const multiply = (n) => n * fac;
  return multiply;
}

const square = (n) => n * n;

const compose = (func1, func2) => {
  return (n) => {
    return func2(func1(n));
  }
}

const myCal = compose(square, multiplyBy);

const fac = 0.1;
const nums = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const transformedNums = nums.map((n) => myCal(n)(fac));
console.log(transformedNums);
